djangosaml2idp/views.py

```diff
@@ -93,7 +110,37 @@ def set_sp(self, sp_entity_id):
         try:
             self.sp['config'] = settings.SAML_IDP_SPCONFIG[sp_entity_id]
         except KeyError:
-            raise ImproperlyConfigured("No config for SP {} defined in SAML_IDP_SPCONFIG".format(sp_entity_id))
+            raise ImproperlyConfigured(_("No config for SP {} defined in SAML_IDP_SPCONFIG").format(sp_entity_id))
+
+    def build_response_arguments(self, req_info):
+        """ Gather response arguments from SAML request
+        """
+        try:
+            self.resp_args = self.IDP.response_args(req_info.message)
+        except (UnknownPrincipal, UnsupportedBinding) as excp:
+            return self.handle_error(request, exception=excp, status=400)
+
+        try:
+            self.set_sp(self.resp_args['sp_entity_id'])
+            self.set_processor()
+        except (KeyError, ImproperlyConfigured) as excp:
+            return self.handle_error(request, exception=excp, status=400)
+
+    def verify_request_signature(self, req_info):
+        """ Signature verification
+            for authn request signature_check is at
+            saml2.sigver.SecurityContext.correctly_signed_authn_request
+        """
+        # TODO: verify that passed signs are the same of sp's metadatas signs
+        verified_ok = req_info.signature_check(req_info.xmlstr)
```

Comment:
Why are you declaring the variable, and not just checking if not req_info.signature_check(req_info.xmlstr)?

Reply:
legacy code, verified_ok comes from it. I Agree, it's strange but I thought the same thing :)

Reply:
Fixed